src/main/scala/view/scalaFX/FXControllers/BaseAppController.scala

```diff
@@ -177,15 +182,19 @@ class BaseAppController(
       simulationPane.children ++= newBunnyViews.map(_.imageView)
       // Start movement of the new bunnies
       newBunnyViews foreach { _.play() }
+      updateClock(GenerationPhaseConstants.REPRODUCTION.name)
     }
 
-    if (generationPhase.phase == WOLVES_PHASE) {
+    if (generationPhase.phase == WolfPhaseConstants.WOLVES_PHASE) {
+      updateClock(GenerationPhaseConstants.WOLVES.name)
       factorsPanelController --> { _.showWolvesEating() }
     }
 
-    if (generationPhase.phase == FOOD_PHASE) {
-      factorsPanelController --> { _.removeWolves() }
-    }
+    if (generationPhase.phase == FoodPhaseConstants.FOOD_PHASE) updateClock(GenerationPhaseConstants.FOOD.name)
+
+    if (generationPhase.phase == TemperaturePhaseConstants.TEMPERATURE_PHASE)
+      updateClock(GenerationPhaseConstants.HIGH_TEMPERATURE.name)
+
```

Comment:
qui è molto brutto che venga richiamato 4 volte updateClock solo per differenziare le fasi, rifattorizza in modo tale che updateClock prenda in input la generationPhase e poi sposti il check su quale sia la fase